src/docs/asciidoc/storage.adoc

```diff
@@ -205,20 +226,110 @@ Config config = new Config();
-                .setPersistentMemoryDirectory("/mnt/optane/data");
+                .setAllocatorType(POOLED);
+
+PersistentMemoryConfig pmemConfig = memoryConfig.getPersistentMemoryConfig()
+                .addDirectoryConfig(new PersistentMemoryDirectoryConfig("/mnt/pmem0", 0))
+                .addDirectoryConfig(new PersistentMemoryDirectoryConfig("/mnt/pmem1", 1));
+
 config.setNativeMemoryConfig(memoryConfig);
 ----
 
-NOTE: Note that integration with Intel(R) Optane(TM) DC
-is supported on Linux operating system and it is for
-Optane DIMMs (not SSDs).
-
-NOTE: To achieve the best performance using
-Intel(R) Optane(TM) DC persistent memory, we recommend to
-use it for IMap, ICache and Near Cache data structures
-with relatively small values up to 1 KB. In this case,
-the performance is similar to that of standard RAM. 
+NOTE: Note that integration with Intel(R) Optane(TM) DC is supported on Linux
+operating system and it is for Optane DIMMs (not SSDs).
+
+===== Allocation Strategies
+
+Since on multi-socket machines there could be multiple persistent memory
+mount points, the memory allocations need to follow an allocation strategy.
+Starting with 4.1, Hazelcast supports two allocation strategies:
+
+* Round-robin allocation strategy
+* NUMA-aware allocation strategy
+
+Hazelcast's memory allocator chooses and statically caches one of them for
+every allocator thread for the entire lifetime of the Hazelcast instance.
+
+====== Round-robin Allocation Strategy
+
+Hazelcast iterates over the configured persistent memory directories and
+makes sure every allocation is done in a different directory then the last.
+This is a best-effort attempt to distribute the allocations evenly on the
+persistent memory DIMMs, which is important from utilization and performance
+point of view as well. This is the default allocation strategy.
+
+====== NUMA-aware Allocation Strategy
+
+The persistent memory modules are mounted in the memory slots just like the
+regular memory modules and sharing the same memory bus. Therefore, the same
+NUMA locality concerns apply to the persistent memory that apply to regular
+memory. This means accessing the persistent memory modules attached to the
+socket on which the current thread runs is cheaper than accessing the
+persistent memory modules attached to a different socket. These are typically
+referenced as NUMA-local and NUMA-remote memories. To achieve the best
+possible performance, Hazelcast implements a NUMA-aware  allocation strategy
+to ensure all persistent memory accesses are local, if certain conditions hold.
+
+To enable this allocation strategy for a certain thread, the thread has to be
+bounded to a single NUMA node, which means the kernel's scheduler makes sure
+the thread can be scheduled only on the CPUs of a single NUMA node. Starting
+with Hazelcast 4.1 this can be done with thread group granularity. For the
+detailed explanation please refer to the thread affinity documentation. What
```

Comment:
As far as I understand, thread affinity will not be documented, please see the discussion that starts here: https://hazelcast.slack.com/archives/C0356BG4R/p1598267300027700

Reply:
In any case, we will need to update this. If thread affinity will be documented, I think we need to link it here, otherwise we should remove this sentence.